src/CodeGen_Hexagon.cpp

```diff
@@ -1286,6 +1286,7 @@ string CodeGen_Hexagon::mattrs() const {
     if (target.has_feature(Halide::Target::HVX_128)) {
         attrs << ",+hvx-double";
```

Comment:
I think this would be clearer if you used the default constructor and added +hvx in an else case to this if statement